### ISSUE TITLE
Branch names with URL reserved chars turn into bad pull request URLs

### ISSUE THREAD

**Issue by MikaelElkiaer** [jesseduffield/lazygit#1485]
**Describe the bug**
Branch names that are not URL friendly cannot be used for creating GitLab merge requests.
(Probably same issue for other providers.)

For instance, having a branch with a `#` will generate the following URL:
`https://MY-GITLAB/MY-REPO/-/merge_requests/new?merge_request[source_branch]=feature/MY-FEATURE#JIRA-ISSUE`

**To Reproduce**
1. Configure lazygit with a GitLab service
2. Create a branch with a reserved character such as `#`
3. Create a pull request for the branch
4. See that the generated URL is not properly escaped and `#` is treated as an anchor

**Expected behavior**
URL with URL encoded branch name.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Arch
 - Lazygit Version: 0.29
 - The last commit id if you built project from sources (run : ```git-rev parse HEAD```): N/A

**Additional context**

**Comment by MikaelElkiaer** [comment]
I guess it would be enough to URL encode the source branch here:
https://github.com/jesseduffield/lazygit/blob/a5719c530a3094aa7f7377f24f3aa5989ffddf6a/pkg/commands/pull_request.go#L46

But I'm affraid I'm not go-capable enough to find a proper library and try it out in a timely manner.

**Comment by mjarkk** [comment]
I have never used # in a branch name is that a gitlab specific thing / does it work on other sites?
If so we should also fix this for the other sites tough i'm not sure what will happen if we url encode the `/` as it changes the url path does that work on all sites?

**Comment by MikaelElkiaer** [comment]
> I have never used # in a branch name is that a gitlab specific thing / does it work on other sites?
> If so we should also fix this for the other sites tough i'm not sure what will happen if we url encode the `/` as it changes the url path does that work on all sites?

The # doesn't do anything as such in the branch name - not like commit messages, e.g. "Something something closes #1485". I just like to carry over the convention.

`/` should not be a problem, as the slash is allowed once you start on your query parameters. I'd say it is perfectly safe to URL encode a query parameter, as that is exactly what URL encoding is for. :)

**Comment by MikaelElkiaer** [comment]
A small experiment confirms that URL encoding everything is no problem, it is actually what GitLab itself does as a response to a push:
```
> git co -b test/something#hashtagged
Switched to a new branch 'test/something#hashtagged'
> git push -u
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote:
remote: To create a merge request for test/something#hashtagged, visit:
remote:   https://MY-GITLAB/MY-REPO/-/merge_requests/new?merge_request%5Bsource_branch%5D=test%2Fsomething%23hashtagged
remote:
To MY-GITLAB:MY-REPO.git
 * [new branch]      test/something#hashtagged -> test/something#hashtagged
Branch 'test/something#hashtagged' set up to track remote branch 'test/something#hashtagged' from 'origin'.
```
